th2_data_services/data.py

```diff
@@ -693,3 +700,6 @@ def clear_cache(self):
         else:
             if self.__is_cache_file_exists():
                 self.__delete_cache()
+
+    def set_metadata(self, metadata: Dict):  # noqa
+        self.metadata = metadata
```

Comment:
It's common mistake to assign list or dict like this

When you do this, you assign a link to the object, but we need a deep copy.

Reply:
Let's also remake to `update_metadata` if you want to do it like a public method